core/src/main/resources/hudson/node_monitors/LabelsMonitor/column.jelly

```diff
@@ -0,0 +1,39 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:s="/lib/form">
+  <style type="text/css">
+  .all_labels {
+    display: none;
+  }
+  .labels:hover .all_labels {
+    display: block;
+  }
+  </style>
+  <td align="right">
+    <div class="labels">
+      <nobr>
+        <j:set var="count" value="0"/>
+        <j:forEach var="entry" items="${data}">
+          <j:if test="${count &lt; from.maxLabelCount || from.maxLabelCount &lt; 0}">
+              <a href="${rootURL}/label/${entry.name}" tooltip="${h.escape(entry.description)}">${entry.name}</a>&amp;#32;
```

Comment:
i prefer `<br>` instead of `&amp;#32;` When you have longer labels it looks terrible. So it makes also more sense to define the max-labels-count. In that case you define 'row-height'

Reply:
you mean to have one label per line? That will also not look so good, as it wastes the space in the rest of the table.
As it is now it will show several labels in one line and if you have more than the max the remaining are shown when you hover over the table cell.

Reply:
try to put this list there any you will see what is better to read. Or create a option to enable list view. So the user can define which scenario is better for him.
![image](https://user-images.githubusercontent.com/89339813/165453194-ea8bea91-fba7-4fc4-a4bd-5f69ba3c7bfb.png)

Reply:
This is how it looks with 5 labels shown
![image](https://user-images.githubusercontent.com/17767050/165983981-349ae851-1698-4be9-80b0-0fad5d4e1c65.png)
with the labels expanded:
![image](https://user-images.githubusercontent.com/17767050/165984202-4f010f91-4e83-4fd6-8e78-0b42307fbb1c.png)

Reply:
With maxlabels = 1 it looks like this when all labels are shown:
![image](https://user-images.githubusercontent.com/17767050/165987337-91d5391c-63c7-4ea7-8573-995f48180676.png)